src/tracker/user.rs

```diff
@@ -99,11 +99,14 @@ impl Map {
                 });
 
             let new_passkey = if let Some(new_passkey) = &user.new_passkey {
-                if let Ok(new_passkey) = Passkey::from_str(new_passkey) {
-                    tracker.passkey2id.write().swap_remove(&passkey);
-                    new_passkey
-                } else {
-                    return StatusCode::BAD_REQUEST;
+                match Passkey::from_str(new_passkey) {
+                    Ok(new_passkey) => {
+                        tracker.passkey2id.write().swap_remove(&passkey);
+                        new_passkey
+                    }
+                    _ => {
+                        return StatusCode::BAD_REQUEST;
+                    }
```

Comment:
I don't think this change is necessary for the 2024 edition for example. It's perfectly fine for the temporary variable to be dropped earlier

Reply:
Good thing I'm reviewing then :joy: